dftimewolf/lib/collectors/grr_hosts.py

```diff
@@ -103,7 +130,21 @@ def _GetClientBySelector(self, selector):
       self.ModuleError('Could not get client for {0:s}'.format(
           selector), critical=True)
 
-    last_seen, client = sorted(result, key=lambda x: x[0], reverse=True)[0]
+    last_month_list = self._ActiveLastMonth(result)
+    if len(last_month_list) >1:
+      self.ModuleError(
+            'Multiple hosts ({0:d}) with the same '
+            'FQDN: "{1:s}" have been active in the last month.\n'
+            'Please use client ID instead of the hostname.'.format(
+                len(last_month_list), selector), critical=True)
+    if not last_month_list:
+      self.ModuleError(
+            'No host with FQDN: "{0:s}" '
+            'has been active in the last month.\n'
+            'Please use client ID instead of the hostname.'.format(
+                selector), critical=True)
```

Comment:
This error message will be displayed to the user if no active clients can be found, regardless of the selector (FQDN or client_id), which will be a bit confusing if the user is already using a Client ID. A better message would be `no 30-day active client found for selector {selector}`

Additionally, it would probably be interesting to specify how many old / inactive clients were found, so that the user can instantly get an idea of whether they typed the FQDN wrong or if Clients were found but not active.

Reply:
done